dependency-analysis/main.go

```diff
@@ -0,0 +1,349 @@
+// Copyright 2023 OpenSSF Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v46/github"
+	"golang.org/x/oauth2"
+)
+
+func main() {
+	vulnerabilities := ""
+	result := ""
+
+	repo := os.Getenv("GITHUB_REPOSITORY")
+	commitSHA := os.Getenv("GITHUB_SHA")
+	token := os.Getenv("GITHUB_TOKEN")
+	pr := os.Getenv("GITHUB_PR_NUMBER")
+	ghUser := os.Getenv("GITHUB_ACTOR")
+	if err := Validate(token, repo, commitSHA, pr); err != nil {
+		log.Fatal(err)
+	}
+
+	ownerRepo := strings.Split(repo, "/")
+	owner := ownerRepo[0]
+	repo = ownerRepo[1]
+	checks, err := GetScorecardChecks()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defaultBranch, err := getDefaultBranch(owner, repo, token)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(context.Background(), ts)
+	client := github.NewClient(tc)
+	data, err := GetDependencyDiff(owner, repo, token, defaultBranch, commitSHA)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	m := make(map[string]DependencyDiff)
+	for _, dep := range *data { //nolint:gocritic
+		m[dep.SourceRepositoryURL] = dep
+	}
+
+	for k, i := range m { //nolint:gocritic
+		url := strings.TrimPrefix(k, "https://")
+		scorecard, err := GetScore(url)
+		if err != nil && len(i.Vulnerabilities) > 0 {
+			sb := strings.Builder{}
+			sb.WriteString(fmt.Sprintf("<details><summary>Vulnerabilties %s</summary>\n </br>", i.SourceRepositoryURL))
+			sb.WriteString("<table>\n")
+			sb.WriteString("<tr>\n")
+			sb.WriteString("<th>Severity</th>\n")
+			sb.WriteString("<th>AdvisoryGHSAId</th>\n")
+			sb.WriteString("<th>AdvisorySummary</th>\n")
+			sb.WriteString("<th>AdvisoryUrl</th>\n")
+			sb.WriteString("</tr>\n")
+			for _, v := range i.Vulnerabilities {
+				sb.WriteString("<tr>\n")
+				sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.Severity))
+				sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.AdvisoryGHSAId))
+				sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.AdvisorySummary))
+				sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.AdvisoryURL))
+			}
+			sb.WriteString("</table>\n")
+			sb.WriteString("</details>\n")
+			vulnerabilities += sb.String()
+			continue
```

Comment:
having a vulnerability will skip all the other checks for a repository? I know having a vuln will cause the action to fail, but is it not worth having the other output?

Reply:
It will continue only when it is an error `GetScorecardResult`.